Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10/MSFT_IntuneDeviceConfigurationIdentityProtectionPolicyWindows10.psm1

```diff
@@ -0,0 +1,1538 @@
+            Save-M365DSCPartialExport -Content $currentDSCBlock `
+                -FileName $Global:PartialExportFileName
+            $i++
+            Write-Host $Global:M365DSCEmojiGreenCheckMark
+        }
+        return $dscContent
+    }
+    catch
+    {
+        Write-Host $Global:M365DSCEmojiRedX
+
+        New-M365DSCLogEntry -Message 'Error during Export:' `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return ''
+    }
+}
+    function Update-DeviceConfigurationPolicyAssignment
+    {
+        [CmdletBinding()]
+        [OutputType([System.Collections.Hashtable])]
+        param (
+            [Parameter(Mandatory = 'true')]
+            [System.String]
+            $DeviceConfigurationPolicyId,
+
+            [Parameter()]
+            [Array]
+            $Targets,
+
+            [Parameter()]
+            [System.String]
+            $Repository='deviceManagement/configurationPolicies',
+
+            [Parameter()]
+            [ValidateSet('v1.0','beta')]
+            [System.String]
+            $APIVersion='beta'
+        )
+        try
+        {
+            $deviceManagementPolicyAssignments=@()
+
+            $Uri="https://graph.microsoft.com/$APIVersion/$Repository/$DeviceConfigurationPolicyId/assign"
+
+            foreach($target in $targets)
+            {
+                $formattedTarget=@{"@odata.type"=$target.dataType}
+                if($target.groupId)
+                {
+                    $formattedTarget.Add('groupId',$target.groupId)
+                }
+                if($target.collectionId)
+                {
+                    $formattedTarget.Add('collectionId',$target.collectionId)
+                }
+                if($target.deviceAndAppManagementAssignmentFilterType)
+                {
+                    $formattedTarget.Add('deviceAndAppManagementAssignmentFilterType',$target.deviceAndAppManagementAssignmentFilterType)
+                }
+                if($target.deviceAndAppManagementAssignmentFilterId)
+                {
+                    $formattedTarget.Add('deviceAndAppManagementAssignmentFilterId',$target.deviceAndAppManagementAssignmentFilterId)
+                }
+                $deviceManagementPolicyAssignments+=@{'target'= $formattedTarget}
+            }
+            $body=@{'assignments'=$deviceManagementPolicyAssignments}|ConvertTo-Json -Depth 20
+            #write-verbose -Message $body
+            Invoke-MgGraphRequest -Method POST -Uri $Uri -Body $body -ErrorAction Stop
+
+        }
+        catch
+        {
+            New-M365DSCLogEntry -Message 'Error updating data:'
+                -Exception $_
+                -Source $($MyInvocation.MyCommand.Source)
+                -TenantId $TenantId
+                -Credential $Credential
+
+            return $null
+        }
+
+
+    }function Rename-M365DSCCimInstanceParameter
```

Comment:
Please place the function on a new line and place an empty line between the closing brace and the function

Reply:
done